deployer/gateway_name_deployer.go

```diff
@@ -3,164 +3,152 @@ package deployer
 	deployments := make(map[uint32]gridtypes.Deployment)
-	var wls []gridtypes.Workload
-
-	err := k.Validate(ctx, sub)
-	if err != nil {
-		return nil, err
-	}
 
-	dl := workloads.NewGridDeployment(k.deployer.twinID, wls)
-	wls = append(wls, k.Gw.ZosWorkload())
-	dl.Workloads = wls
-	deployments[k.Node] = dl
+	dl := workloads.NewGridDeployment(k.tfPluginClient.TwinID, []gridtypes.Workload{})
+	dl.Workloads = append(dl.Workloads, gw.ZosWorkload())
 
+	deployments[gw.NodeID] = dl
 	return deployments, nil
 }
 
-func (k *GatewayNameDeployer) InvalidateNameContract(ctx context.Context, sub subi.SubstrateExt) (err error) {
-	if k.NameContractID == 0 {
+func (k *GatewayNameDeployer) InvalidateNameContract(ctx context.Context, gw *workloads.GatewayNameProxy) (err error) {
+	if gw.NameContractID == 0 {
 		return
 	}
 
-	k.NameContractID, err = sub.InvalidateNameContract(
+	gw.NameContractID, err = k.tfPluginClient.SubstrateConn.InvalidateNameContract(
 		ctx,
-		k.TFPluginClient.Identity,
-		k.NameContractID,
-		k.Gw.Name,
+		k.tfPluginClient.Identity,
+		gw.NameContractID,
+		gw.Name,
 	)
 	return
 }
 
-func (k *GatewayNameDeployer) Deploy(ctx context.Context, sub subi.SubstrateExt) error {
-	if err := k.Validate(ctx, sub); err != nil {
+func (k *GatewayNameDeployer) Deploy(ctx context.Context, gw *workloads.GatewayNameProxy) error {
+	if err := k.Validate(ctx, gw); err != nil {
 		return err
 	}
-	newDeployments, err := k.GenerateVersionlessDeployments(ctx, sub)
+	newDeployments, err := k.GenerateVersionlessDeployments(ctx, gw)
 	if err != nil {
 		return errors.Wrap(err, "couldn't generate deployments data")
 	}
 
 	deploymentData := workloads.DeploymentData{
-		Name: k.Gw.Name,
+		Name: gw.Name,
 		Type: "Gateway Name",
 	}
 	newDeploymentsData := make(map[uint32]workloads.DeploymentData)
 	newDeploymentsSolutionProvider := make(map[uint32]*uint64)
 
-	newDeploymentsData[k.Node] = deploymentData
-	newDeploymentsSolutionProvider[k.Node] = nil
+	newDeploymentsData[gw.NodeID] = deploymentData
+	newDeploymentsSolutionProvider[gw.NodeID] = nil
 
-	if err := k.InvalidateNameContract(ctx, sub); err != nil {
+	if err := k.InvalidateNameContract(ctx, gw); err != nil {
 		return err
 	}
-	if k.NameContractID == 0 {
-		k.NameContractID, err = sub.CreateNameContract(k.TFPluginClient.Identity, k.Gw.Name)
+	if gw.NameContractID == 0 {
+		gw.NameContractID, err = k.tfPluginClient.SubstrateConn.CreateNameContract(k.tfPluginClient.Identity, gw.Name)
 		if err != nil {
 			return err
 		}
 	}
-	if k.ID == "" {
-		// create the resource if the contract is created
-		k.ID = uuid.New().String()
-	}
-	k.NodeDeploymentID, err = k.deployer.Deploy(ctx, k.NodeDeploymentID, newDeployments, newDeploymentsData, newDeploymentsSolutionProvider)
+	gw.NodeDeploymentID, err = k.deployer.Deploy(ctx, gw.NodeDeploymentID, newDeployments, newDeploymentsData, newDeploymentsSolutionProvider)
```

Comment:
here we need to update state and get old deployments from state loader
`oldDeployments := k.tfPluginClient.StateLoader.currentNodeDeployment`
and not using gw.NodeDeploymentID